packages/fast-color-explorer/app/color-blocks.tsx

```diff
@@ -0,0 +1,581 @@
+                            backgroundColor: neutralForegroundRest(
+                                this.state.designSystem
+                            ),
+                        })
+                    )}
+                    state={SwatchStates.rest}
+                    showTextContrast={false}
+                />
+
+                <Caption jssStyleSheet={this.titleStyleOverrides}>
+                    NEUTRAL / FOREGROUND / HINT
+                </Caption>
+                <Swatch
+                    color={neutralForegroundHint(this.state.designSystem)}
+                    backgroundColor={this.state.designSystem.backgroundColor}
+                    textColor={neutralForegroundRest(
+                        Object.assign({}, this.state.designSystem, {
+                            backgroundColor: neutralForegroundHint(
+                                this.state.designSystem
+                            ),
+                        })
+                    )}
+                    state={SwatchStates.rest}
+                    showTextContrast={false}
+                />
+
+                <Caption jssStyleSheet={this.titleStyleOverrides}>
+                    ACCENT / FOREGROUND
+                </Caption>
+                <Swatch
+                    color={accentColors.rest}
+                    backgroundColor={this.state.designSystem.backgroundColor}
+                    textColor={accentForegroundCut(
+                        (designSystem: ColorsDesignSystem): string => accentColors.rest
+                    )(this.state.designSystem)}
+                    state={SwatchStates.rest}
+                    showTextContrast={false}
+                />
+                <Swatch
+                    color={accentColors.hover}
+                    backgroundColor={this.state.designSystem.backgroundColor}
+                    textColor={accentForegroundCut(
+                        (designSystem: ColorsDesignSystem): string => accentColors.hover
+                    )(this.state.designSystem)}
+                    state={SwatchStates.hover}
+                    showTextContrast={false}
+                />
+                <Swatch
+                    color={accentColors.active}
+                    backgroundColor={this.state.designSystem.backgroundColor}
+                    textColor={accentForegroundCut(
+                        (designSystem: ColorsDesignSystem): string => accentColors.active
+                    )(this.state.designSystem)}
+                    state={SwatchStates.active}
+                    showTextContrast={false}
+                />
+            </React.Fragment>
+        );
+    }
+}
+
+/* tslint:disable-next-line */
+const ColorBlocks = manageJss(styles)(ColorBlocksBase);
+type ColorBlocks = InstanceType<typeof ColorBlocks>;
+
+function mapStateToProps(state: AppState): Partial<ColorBlocksProps> {
+    return {
+        component: state.componentType,
+        designSystem: state.designSystem,
+    };
```

Comment:
could just do `return { ...state };` if you matched the props to state for component/componentType